### DISCUSSION TITLE
Various debuff effects: slow, weakness etc

### DISCUSSION THREAD

**Discussion by Kiwi1444** [ikemen-engine/Ikemen-GO#2197] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

It would be nice to implement various penalty effects with the possibility of specifying the time of action. For example:

**Slowdown**

-Reduces both horizontal and vertical movement speed (it is possible to specify the percentage of speed reduction). Or horizontal and vertical movement speed separately.

-Ability to slow down animation (all actions including attacks will be performed slower).

**Weakness**

-Reduces defense and attack both together and separately.

-Ability to specify the percentage of reduction

**Confusion**

Total or partial loss of control. Inability to perform certain actions (such as defending or attacking)

And so on.

Graphically show that the character is under this or that effect (palette change, blinking, special icon above the head)

**Comment by potsmugen** [comment, marked as answer]
You can already do all of those with varying levels of success. An engine should give you the pieces to create your own codes, not the codes themselves.

**Comment by Kiwi1444** [comment]
I was thinking about how to limit opponent's movement speed without sending him into custom state or slowing down his animation. It turns out that you need to bind a helper to him, limiting his movements from three sides. And the collision boxes should "outside" the opponent. But it sounds so irrational..... Instead of putting a temporary debuff, you have to invent a badly working bicycle.
